### PR TITLE
NOTIF-482 Allow devs to delete their own app

### PR DESCRIPTION
**Pull request by josejulio**
Allows apps to delete their own applications.

I'm unsure if we should allow this permission, but lets try on stage first for some time.

### CODE REVIEW

File: backend/src/test/java/com/redhat/cloud/notifications/routers/InternalServiceTest.java [hunk 1]
```diff
@@ -389,6 +389,7 @@ void testVersion() {
         assertTrue(responseBody.matches("^[0-9a-f]{7}$"));
     }
 
+    @Test
```
Comment (josejulio) [inline backend/src/test/java/com/redhat/cloud/notifications/routers/InternalServiceTest.java:392]: Can't believe I missed this

Comment (gwenneg) [reply backend/src/test/java/com/redhat/cloud/notifications/routers/InternalServiceTest.java:392]: 😱

**Comment by gwenneg** [comment]
> I'm unsure if we should allow this permission, but lets try on stage first for some time.

Yeah... I'm not sure this is a good idea unless the application deletion is a soft delete (based on a boolean in the DB).

**Comment by josejulio** [comment]
I'll close it for now - will send a separate PR about the missing `@Test` annotation. We can decide later how we want to handle the soft deletion.

**Pull request closed without merging**